### PR TITLE
Travis/heroku

### PR DESCRIPTION
**Pull request by festinalente91**
Add travis-heroku continuous deployment

https://shapeshed.com/continuously-deploy-node-apps-with-github-travis-and-heroku/

**Issue**

Heroku run app with production mode so devDependencies are not installed.
I moved related modules from "devDependencies" to "dependencies".
If you want to move some modules to devDependencies please tell.

### CODE REVIEW

File: gulpfile.babel.js [hunk 1]
```diff
@@ -98,7 +98,7 @@ gulp.task('watch', ['build', 'apidoc'], () => {
   })
 });
 
-gulp.task('server', ['build'], () => {
+gulp.task('server', ['build', 'apidoc'], () => {
```
Comment (ovekyc) [inline gulpfile.babel.js]: We should drop it later, in REAL product but now okay

**Comment by ovekyc** [comment]
LGTM :) heroku master

File: gulpfile.babel.js (hunk 1 above)
Comment (festinalente91) [inline gulpfile.babel.js]: Right. We need to keep tracking it.

**Comment by fguy** [comment]
Please revert devDependencies. See example project how it deal with build script.

**Comment by festinalente91** [comment]
@fguy I tried to solve this problem with script. but there is a time limit in heroku :(

heroku's NODE_ENV is production so only install 'dependencies' 
=> I need to install 'devDependencies' before 'npm run server' so I change the script like
'npm install --only=dev && npm run server' 
=> And it worked, but there was time limit in heroku so I couldn't deploy.
=> So I chose to move devDependencies to dependencies (which is not time counted, because heroku npm install in production ENV before start Procfile script).

File: .travis.yml [hunk 2]
```diff
@@ -1,9 +1,14 @@
...
+- npm install
+- echo '{}' > config/local.json
 script: npm test
 notifications:
   slack: siliconvalleybootcamp:SnRpv9p0fK5FmSEFa7iVqJAw
+deploy:
+  app: goober
+  provider: heroku
+  api_key:
+    secure: I0UNPhpj2iX8QZeyfRp+PrazFFa7Jfbg4e2Nz/LvgY2JLdpjqoglXBG5JjBD303ZBueEA+odjFsdOFqAmUDiOb1VShGmBZ/tp+Msq7AnqWxv8siS+bZcVyRi7SV6cPR9DHxUi3e4Y/XFovh1zY0Mc7npe1H6/QP0Z+UUzzWmQrRWPhWr6F4Z6xEPvFzoQN9yzbs/YBp+JeL5ZD5y62eghzP4MgwcbaBLcqVN9Ux6uhlEd3OouK3b/08Os1ApBgkVsa1/BcDjATl5RHfagPeK7M3OZ1nUt7ahiCRpitIEytY5eF08uqtSwk1GE0jnfFOe6CUpNtamUet29914/3GPWeUi3zd+zwuqpazKXAJUmCpSBhLZju5+9IcwTd4YUHzbWs5jo4Sa8Z/GzeyH1PiDXqprqhhYWRVD5NtXMpZa1XkKQBZFLb6woZU+B5PInPa5ndMphcGzhWX399+9nB7tGAX967L+Y2F9/0sbjhtT+/+OUriemj8x36LU1q5a20wSJW4scaLzyzTIfTr8uDoeX2N75VE0yw12fHLg/8xnnLkWzVMYYXNBE3V4Yhz8CMCyUL/abDnH31wIQNhI/WliJeR98nL+mGXBhuE59cSTi5D6o6TMdGKXBrGTwyzpGmnToNezg6eKVKYA/917PG5t+O+VKZhyxsnAcACV4NQ6ZLo=
```
Comment (fguy) [inline .travis.yml]: Do not include this information in the repo.  Please discard this key immediately.

File: .travis.yml [hunk 3]
```diff
@@ -1,9 +1,14 @@
 language: node_js
 node_js:
-  - "6"
+- '6'
 before_script:
-  - npm install
-  - echo '{}' > config/local.json
+- npm install
+- echo '{}' > config/local.json
```
Comment (fguy) [inline .travis.yml]: You do not need this. My mistake on example-project.

**Comment by fguy** [comment]
@festinalente91 That's not a right solution. Conceptually, devDependencies are not dependencies. Please find another solution. Something's wrong if it takes too long time to build.

**Comment by festinalente91** [comment]
@fguy  okay I'll check again and find out other ways :)

**Pull request closed without merging**